src/index.js

```diff
@@ -90,7 +90,7 @@ const shouldRedirect = (decodedPath, {redirects = [], trailingSlash}, cleanUrl)
 	}
 
 	const defaultType = 301;
-	const matchHTML = /.html|.htm|\/index/g;
+	const matchHTML = /.html|.htm|\/index$/g;
```

Comment:
@timneutkens I've considered this - but why would a path contain `.htm`, if not at the end?

Reply:
Ok, let's just support it anyway.

Reply:
It would just lead to unexpected behavior, I've seen the same happen with Next.js where we had a bug where the regex for matching externals was `/.js/` and it matches paths like `/highlight.js/index.js`. And your intent is to match the ending of the string, so it should match the ending of the string 🕵️